src/core/search/indices.h

```diff
@@ -205,4 +206,29 @@ struct HnswVectorIndex : public BaseVectorIndex {
   std::unique_ptr<HnswlibAdapter> adapter_;
 };
 
+struct GeoIndex : public BaseIndex {
+  using point =
+      boost::geometry::model::point<double, 2,
+                                    boost::geometry::cs::geographic<boost::geometry::degree>>;
+  using index_entry = std::pair<point, DocId>;
+
+  explicit GeoIndex(PMR_NS::memory_resource* mr);
+  ~GeoIndex();
+
+  bool Add(DocId id, const DocumentAccessor& doc, std::string_view field) override;
+  void Remove(DocId id, const DocumentAccessor& doc, std::string_view field) override;
+  std::vector<DocId> RadiusSearch(double lat, double lon, double radius) const;
+
+  std::vector<DocId> GetAllDocsWithNonNullValues() const override {
+    return std::vector<DocId>{};
+  }
```

Comment:
That need to be for a `@geofield:*` query? Is this possible? If not, we should explicitly disallow it. Not saying to solve it now, just a TODO would help us not to forget

Reply:
Added function to return all documents